src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -2046,3 +2047,193 @@ def get_custom_locations_oid(cmd, cl_oid):
+
+            # Check for msi certificate
+            if all_agents_stuck is False:
+                diagnostic_checks["msi_cert_check"] = troubleshootutils.check_msi_certificate_presence(corev1_api_instance)
+
+            # If msi certificate present then only we will perform msi certificate expiry check
+            if diagnostic_checks["msi_cert_check"] == consts.Diagnostic_Check_Passed:
+                diagnostic_checks["msi_cert_expiry_check"] = troubleshootutils.check_msi_expiry(connected_cluster)
+
+            # If msi certificate present then only we will do Kube aad proxy checks
+            if diagnostic_checks["msi_cert_check"] == consts.Diagnostic_Check_Passed:
+                diagnostic_checks["kap_security_policy_check"] = troubleshootutils.check_cluster_security_policy(corev1_api_instance, helm_client_location, release_namespace)
+
+                # If no security policy is present in cluster then we can check for the Kube aad proxy certificate
+                if diagnostic_checks["kap_security_policy_check"] == consts.Diagnostic_Check_Passed:
+                    diagnostic_checks["kap_cert_check"] = troubleshootutils.check_kap_cert(corev1_api_instance)
+
+            # Checking whether optional extra values file has been provided.
+            values_file_provided, values_file = utils.get_values_file()
+
+            # Validate the helm environment file for Dogfood.
+            dp_endpoint_dogfood = None
+            release_train_dogfood = None
+            if cmd.cli_ctx.cloud.endpoints.resource_manager == consts.Dogfood_RMEndpoint:
+                consts.Azure_DogfoodCloudName
+                dp_endpoint_dogfood, release_train_dogfood = validate_env_file_dogfood(values_file, values_file_provided)
+
+            # Adding helm repo
+            if os.getenv('HELMREPONAME') and os.getenv('HELMREPOURL'):
+                utils.add_helm_repo(kube_config, kube_context, helm_client_location)
+
+            # Setting the config dataplane endpoint
+            config_dp_endpoint = get_config_dp_endpoint(cmd, connected_cluster.location)
+
+            # Retrieving Helm chart OCI Artifact location
+            registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood, release_train_dogfood)
+
+            # Get azure-arc agent version for telemetry
+            azure_arc_agent_version = registry_path.split(':')[1]
+
+            # Check for agent verison comaptibility
+            diagnostic_checks["agent_version_check"] = troubleshootutils.check_agent_version(connected_cluster, azure_arc_agent_version)
+        else:
+            print("Error : Arc Agents have not been added to cluster. Please delete and reconnect the Kubernetes Cluster\n")
```

Comment:
How about this  - Arc agents are not present on the cluster. Please verify if Arc onboarding of the kubernetes cluster has been attempted.